node/src/main/scala/coop/rchain/node/configuration/Configuration.scala

```diff
@@ -79,7 +81,14 @@ object Configuration {
     val mergedConf = optionsConfig
       .withFallback(fileConfig)
       .withFallback(defaultConfig)
-    val nodeConfE = mergedConf.load[NodeConf]
+
+    val nodeConfE = if (options.run.posMultiSigPublicKeys.isSupplied) {
+      implicit val listOfStringsReader =
+        ConfigReader.fromString[List[String]](_.split(" ").toList.asRight)
+      mergedConf.load[NodeConf]
+    } else {
+      mergedConf.load[NodeConf]
+    }
```

Comment:
This solution is not elegant, but here is the reason why I implemented it this way.

The problem is that `ConfigValue` can only be a primitive type. `List` doesn't fit:
```
bug in method caller: not valid to create ConfigValue from: List(a, b, c)
```

Therefore, `List[String]` must be converted to a `String`. I am using space as delimiter.
After [concatenating the configs](https://github.com/rchain/rchain/blob/5d7d53ff6b38ec3d093d3d3410a3775a0f488db6/node/src/main/scala/coop/rchain/node/configuration/Configuration.scala#L81-L83), `pureconfig` tries to populate the [`posMultiSigPublicKeys`](https://github.com/rchain/rchain/blob/ac73fa0f7d1f21e4aa4ec24a1a55d6a413e65edc/casper/src/main/scala/coop/rchain/casper/CasperConf.scala#L33) field, the type of which is `List[String]`.
Among the implicit, a suitable converter is selected. But for defaults.conf and CLI, it should be different, since in the first case it is a list of strings, and in the second it is a value combined into one line.

Reply:
This seems like a reasonable solution. Is there a reason to only use definition of the list reader just for this case or to define it together with other custom readers?

Reply:
I don't fully understand, but if I move the implicit above to other implicits, the configuration stops loading if the `pos-multi-sig-public-keys` parameter is not set or empty. Here is error message
```
Can't build the configuration: 
ConvertFailure(WrongType(LIST,Set(STRING)),Some(ConfigOrigin(defaults.conf @ file:<path-to-project-repo>/node/target/scala-2.12/classes/defaults.conf)),casper.genesis-block-data.pos-multi-sig-public-keys)
```
It looks like this implicit is being selected for reading instead of the right one.